File: record-generator/processor/src/main/java/me/bristermitten/mittenlib/codegen/union/UnionSpec.java
package me.bristermitten.mittenlib.codegen.union;

import com.squareup.javapoet.ClassName;
import java.util.List;
import me.bristermitten.mittenlib.codegen.GenericTypeSpec;
import me.bristermitten.mittenlib.codegen.MatchStrategies;
import me.bristermitten.mittenlib.codegen.record.RecordConstructorSpec;

public record UnionSpec(
        ClassName source, ClassName name, MatchStrategies strategy, List<RecordConstructorSpec> constructors)
        implements GenericTypeSpec {}

File: record-generator/processor/src/main/scala/me/bristermitten/mittenlib/codegen/union/UnionSpec.scala
package me.bristermitten.mittenlib.codegen.union

import com.palantir.javapoet.ClassName
import java.util.List
import me.bristermitten.mittenlib.codegen.GenericTypeSpec
import me.bristermitten.mittenlib.codegen.MatchStrategies
import me.bristermitten.mittenlib.codegen.record.RecordConstructorSpec

case class UnionSpec(
  source: ClassName,
  name: ClassName,
  strategy: MatchStrategies,
  constructors: List[RecordConstructorSpec]
) extends GenericTypeSpec
